File: scripts/organize_assets.js
const fs = require('fs');
const path = require('path');

const baseDir = path.resolve(__dirname, '..');
const assestsDir = path.join(baseDir, 'Assests');
const publicDir = path.join(baseDir, 'public');
const publicAssets = path.join(publicDir, 'assets');

const subdirs = [
  'profile',
  'projects/agmis',
  'projects/ai-voice-agent',
  'projects/doc-summarizer',
  'projects/spam-email-classifier',
  'projects/n8n-restaurant-bot',
  'projects/natural-voice-rag',
  'projects/code-explainer',
  'projects/code-comments-generator',
  'projects/sentiment-analysis',
  'projects/ai-system-auditor',
  'projects/archaeo-mind',
  'projects/skilmind-ai',
  'projects/snapchat-organizer',
  'certifications',
  'credentials',
  'events',
  'diagrams',
  'social',
  'general',
  'logo',
  'resume'
];

subdirs.forEach(sd => {
  fs.mkdirSync(path.join(publicAssets, sd), { recursive: true });
});

const manifestItems = [];

function copyAndManifest(srcDir, dstSubdir, projectName, usageDesc) {
  if (!fs.existsSync(srcDir)) return;
  const files = fs.readdirSync(srcDir);
  files.forEach(fname => {
    const srcFile = path.join(srcDir, fname);
    if (fs.statSync(srcFile).isFile()) {
      const dstFile = path.join(publicAssets, dstSubdir, fname);
      fs.copyFileSync(srcFile, dstFile);
      const relPath = `/assets/${dstSubdir}/${fname}`.replace(/\\/g, '/');
      manifestItems.push({
        filename: fname,
        path: relPath,
        source: 'Local Asset Audit',
        project: projectName,
        usage: usageDesc,
        format: path.extname(fname).replace('.', '').toLowerCase()
      });
    }
  });
}

const projectsSrc = path.join(assestsDir, 'Projects');

// AGMIS
copyAndManifest(path.join(projectsSrc, 'Agmis'), 'projects/agmis', 'AGMIS', 'Academic dashboard and analytical views');
fs.mkdirSync(path.join(publicAssets, 'agmis'), { recursive: true });
if (fs.existsSync(path.join(projectsSrc, 'Agmis'))) {
  fs.readdirSync(path.join(projectsSrc, 'Agmis')).forEach(f => {
    const s = path.join(projectsSrc, 'Agmis', f);
    if (fs.statSync(s).isFile()) fs.copyFileSync(s, path.join(publicAssets, 'agmis', f));
  });
}

// Voice Agent
copyAndManifest(path.join(projectsSrc, 'Ai voice agent'), 'projects/ai-voice-agent', 'AI Voice Agent', 'Voice agent interface and workflow previews');
fs.mkdirSync(path.join(publicAssets, 'voice-agent'), { recursive: true });
if (fs.existsSync(path.join(projectsSrc, 'Ai voice agent'))) {
  fs.readdirSync(path.join(projectsSrc, 'Ai voice agent')).forEach(f => {
    const s = path.join(projectsSrc, 'Ai voice agent', f);
    if (fs.statSync(s).isFile()) fs.copyFileSync(s, path.join(publicAssets, 'voice-agent', f));
  });
}

// Doc Summarizer
copyAndManifest(path.join(projectsSrc, 'Document summrizer'), 'projects/doc-summarizer', 'Huntii Analyser', 'Multi-document analysis interface screenshots');
fs.mkdirSync(path.join(publicAssets, 'doc-summarizer'), { recursive: true });
if (fs.existsSync(path.join(projectsSrc, 'Document summrizer'))) {
  fs.readdirSync(path.join(projectsSrc, 'Document summrizer')).forEach(f => {
    const s = path.join(projectsSrc, 'Document summrizer', f);
    if (fs.statSync(s).isFile()) fs.copyFileSync(s, path.join(publicAssets, 'doc-summarizer', f));
  });
}

// n8n Automation
copyAndManifest(path.join(projectsSrc, 'n8n-Automation'), 'projects/n8n-restaurant-bot', 'n8n AI Restaurant Agent', 'n8n workflow and canvas screenshots');
fs.mkdirSync(path.join(publicAssets, 'n8n-restaurant-bot'), { recursive: true });
if (fs.existsSync(path.join(projectsSrc, 'n8n-Automation'))) {
  fs.readdirSync(path.join(projectsSrc, 'n8n-Automation')).forEach(f => {
    const s = path.join(projectsSrc, 'n8n-Automation', f);
    if (fs.statSync(s).isFile()) fs.copyFileSync(s, path.join(publicAssets, 'n8n-restaurant-bot', f));
  });
}

// Spam Email Classifier
copyAndManifest(path.join(projectsSrc, 'spam-email-classifier'), 'projects/spam-email-classifier', 'Spam Classifier', 'Spam classifier UI and terminal benchmarks');
fs.mkdirSync(path.join(publicAssets, 'spam-email-classifier'), { recursive: true });
if (fs.existsSync(path.join(projectsSrc, 'spam-email-classifier'))) {
  fs.readdirSync(path.join(projectsSrc, 'spam-email-classifier')).forEach(f => {
    const s = path.join(projectsSrc, 'spam-email-classifier', f);
    if (fs.statSync(s).isFile()) fs.copyFileSync(s, path.join(publicAssets, 'spam-email-classifier', f));
  });
}

// Profile
const profSrc = path.join(assestsDir, 'profile', 'Passport size photo.jpg');
const profDst = path.join(publicAssets, 'profile', 'profile-main.jpg');
if (fs.existsSync(profSrc)) {
  fs.copyFileSync(profSrc, profDst);
  fs.copyFileSync(profSrc, path.join(publicAssets, 'profile', 'Passport size photo.jpg'));
  manifestItems.push({
    filename: 'profile-main.jpg',
    path: '/assets/profile/profile-main.jpg',
    source: 'Professional Headshot',
    project: 'Personal Profile',
    usage: 'Primary portfolio and about page profile picture',
    format: 'jpg'
  });
}

// Logo
const logoSrc = path.join(assestsDir, 'logo', 'logo.png');
if (fs.existsSync(logoSrc)) {
  fs.copyFileSync(logoSrc, path.join(publicAssets, 'logo', 'logo.png'));
  fs.copyFileSync(logoSrc, path.join(publicDir, 'logo.png'));
  manifestItems.push({
    filename: 'logo.png',
    path: '/assets/logo/logo.png',
    source: 'Brand Logo',
    project: 'Personal Brand',
    usage: 'Website header branding logo',
    format: 'png'
  });
}

// Resume
const resumeSrcDir = path.join(assestsDir, 'Resume');
const publicResumeDir = path.join(publicDir, 'resume');
fs.mkdirSync(publicResumeDir, { recursive: true });
if (fs.existsSync(resumeSrcDir)) {
  fs.readdirSync(resumeSrcDir).forEach(f => {
    const sFile = path.join(resumeSrcDir, f);
    if (fs.statSync(sFile).isFile()) {
      fs.copyFileSync(sFile, path.join(publicResumeDir, f));
      fs.copyFileSync(sFile, path.join(publicResumeDir, f.replace(/ /g, '_')));
    }
  });
}

// Accomplishment & Experience -> Credentials
const accDir = path.join(assestsDir, 'Accomplishment');
const credDir = path.join(publicAssets, 'credentials');
fs.mkdirSync(credDir, { recursive: true });
if (fs.existsSync(accDir)) {
  fs.readdirSync(accDir).forEach(f => {
    const sFile = path.join(accDir, f);
    if (fs.statSync(sFile).isFile()) {
      fs.copyFileSync(sFile, path.join(credDir, f));
      const cleanF = f.replace(/ /g, '_').replace(/:/g, '').replace(/-/g, '_');
      fs.copyFileSync(sFile, path.join(credDir, cleanF));
      manifestItems.push({
        filename: f,
        path: `/assets/credentials/${f}`,
        source: 'Local Asset Audit',
        project: 'Credentials & Certificates',
        usage: 'Verified professional credentials and badges',
        format: path.extname(f).replace('.', '').toLowerCase()
      });
    }
  });
}

const expInternDir = path.join(assestsDir, 'Exprience', 'Artificial Intelligence Intern');
if (fs.existsSync(expInternDir)) {
  const certSrc = path.join(expInternDir, 'Internship Certificate.pdf');
  const offerSrc = path.join(expInternDir, 'OFFER LETTER.pdf');
  if (fs.existsSync(certSrc)) {
    fs.copyFileSync(certSrc, path.join(credDir, 'Codec_Technologies_AI_Intern_Certificate.pdf'));
  }
  if (fs.existsSync(offerSrc)) {
    fs.copyFileSync(offerSrc, path.join(credDir, 'Codec_Technologies_AI_Intern_Offer_Letter.pdf'));
  }
}

// Write asset-manifest.json
const manifestPath = path.join(publicAssets, 'asset-manifest.json');
fs.writeFileSync(manifestPath, JSON.stringify({ total_assets: manifestItems.length, assets: manifestItems }, null, 2));

console.log(`Asset Manifest generated with ${manifestItems.length} assets.`);

File: scripts/organize_assets.py
import os
import shutil
import json

base_dir = r"c:\Users\hamsi\OneDrive\Desktop\portfolio-website-build"
assests_dir = os.path.join(base_dir, "Assests")
public_dir = os.path.join(base_dir, "public")
public_assets = os.path.join(public_dir, "assets")

subdirs = [
    "profile",
    "projects/agmis",
    "projects/ai-voice-agent",
    "projects/doc-summarizer",
    "projects/spam-email-classifier",
    "projects/n8n-restaurant-bot",
    "projects/natural-voice-rag",
    "projects/code-explainer",
    "projects/code-comments-generator",
    "projects/sentiment-analysis",
    "projects/ai-system-auditor",
    "projects/archaeo-mind",
    "projects/skilmind-ai",
    "projects/snapchat-organizer",
    "certifications",
    "credentials",
    "events",
    "diagrams",
    "social",
    "general",
    "logo",
    "resume"
]

for sd in subdirs:
    os.makedirs(os.path.join(public_assets, sd), exist_ok=True)

manifest_items = []

def copy_and_manifest(src_dir, dst_subdir, project_name, usage_desc):
    if not os.path.exists(src_dir):
        return
    for fname in os.listdir(src_dir):
        src_file = os.path.join(src_dir, fname)
        if os.path.isfile(src_file):
            dst_file = os.path.join(public_assets, dst_subdir, fname)
            shutil.copy2(src_file, dst_file)
            rel_path = f"/assets/{dst_subdir}/{fname}".replace("\\", "/")
            manifest_items.append({
                "filename": fname,
                "path": rel_path,
                "source": "Local Asset Audit",
                "project": project_name,
                "usage": usage_desc,
                "format": fname.split(".")[-1].lower()
            })

# Helper to sync files to multiple target directories
def sync_to_dirs(src_dir, target_dirs, project_name, usage_desc):
    if not os.path.exists(src_dir):
        return
    for fname in os.listdir(src_dir):
        src_file = os.path.join(src_dir, fname)
        if os.path.isfile(src_file):
            for tdir in target_dirs:
                os.makedirs(tdir, exist_ok=True)
                shutil.copy2(src_file, os.path.join(tdir, fname))
            # Register in manifest once
            rel_path = f"/assets/{os.path.relpath(target_dirs[0], public_assets)}/{fname}".replace("\\", "/")
            manifest_items.append({
                "filename": fname,
                "path": rel_path,
                "source": "Local Asset Audit",
                "project": project_name,
                "usage": usage_desc,
                "format": fname.split(".")[-1].lower()
            })

# Projects mapping
projects_src = os.path.join(assests_dir, "Projects")

# AGMIS
sync_to_dirs(
    os.path.join(projects_src, "Agmis"),
    [
        os.path.join(public_assets, "agmis"),
        os.path.join(public_assets, "Agmis"),
        os.path.join(public_assets, "projects", "agmis"),
        os.path.join(public_assets, "projects", "Agmis"),
    ],
    "AGMIS",
    "Academic dashboard and analytical views"
)

# Voice Agent
sync_to_dirs(
    os.path.join(projects_src, "Ai voice agent"),
    [
        os.path.join(public_assets, "voice-agent"),
        os.path.join(public_assets, "ai-voice-agent"),
        os.path.join(public_assets, "Ai voice agent"),
        os.path.join(public_assets, "projects", "voice-agent"),
        os.path.join(public_assets, "projects", "ai-voice-agent"),
        os.path.join(public_assets, "projects", "Ai voice agent"),
    ],
    "AI Voice Agent",
    "Voice agent interface and workflow previews"
)

# Doc Summarizer
sync_to_dirs(
    os.path.join(projects_src, "Document summrizer"),
    [
        os.path.join(public_assets, "doc-summarizer"),
        os.path.join(public_assets, "document-summarizer"),
        os.path.join(public_assets, "Document summrizer"),
        os.path.join(public_assets, "projects", "doc-summarizer"),
        os.path.join(public_assets, "projects", "document-summarizer"),
        os.path.join(public_assets, "projects", "Document summrizer"),
    ],
    "Huntii Analyser",
    "Multi-document analysis interface screenshots"
)

# n8n Automation
sync_to_dirs(
    os.path.join(projects_src, "n8n-Automation"),
    [
        os.path.join(public_assets, "n8n-restaurant-bot"),
        os.path.join(public_assets, "n8n-automation"),
        os.path.join(public_assets, "n8n-Automation"),
        os.path.join(public_assets, "projects", "n8n-restaurant-bot"),
        os.path.join(public_assets, "projects", "n8n-automation"),
        os.path.join(public_assets, "projects", "n8n-Automation"),
    ],
    "n8n AI Restaurant Agent",
    "n8n workflow and canvas screenshots"
)

# Spam Email Classifier
sync_to_dirs(
    os.path.join(projects_src, "spam-email-classifier"),
    [
        os.path.join(public_assets, "spam-email-classifier"),
        os.path.join(public_assets, "spam-classifier"),
        os.path.join(public_assets, "projects", "spam-email-classifier"),
        os.path.join(public_assets, "projects", "spam-classifier"),
    ],
    "Spam Classifier",
    "Spam classifier UI and terminal benchmarks"
)

# Profile
prof_src = os.path.join(assests_dir, "profile", "Passport size photo.jpg")
prof_dst = os.path.join(public_assets, "profile", "profile-main.jpg")
if os.path.exists(prof_src):
    shutil.copy2(prof_src, prof_dst)
    shutil.copy2(prof_src, os.path.join(public_assets, "profile", "Passport size photo.jpg"))
    manifest_items.append({
        "filename": "profile-main.jpg",
        "path": "/assets/profile/profile-main.jpg",
        "source": "Professional Headshot",
        "project": "Personal Profile",
        "usage": "Primary portfolio and about page profile picture",
        "format": "jpg"
    })

# Logo
logo_src = os.path.join(assests_dir, "logo", "logo.png")
if os.path.exists(logo_src):
    shutil.copy2(logo_src, os.path.join(public_assets, "logo", "logo.png"))
    shutil.copy2(logo_src, os.path.join(public_dir, "logo.png"))
    manifest_items.append({
        "filename": "logo.png",
        "path": "/assets/logo/logo.png",
        "source": "Brand Logo",
        "project": "Personal Brand",
        "usage": "Website header branding logo",
        "format": "png"
    })

# Resume
resume_src_dir = os.path.join(assests_dir, "Resume")
public_resume_dir = os.path.join(public_dir, "resume")
os.makedirs(public_resume_dir, exist_ok=True)
if os.path.exists(resume_src_dir):
    for f in os.listdir(resume_src_dir):
        s_file = os.path.join(resume_src_dir, f)
        if os.path.isfile(s_file):
            shutil.copy2(s_file, os.path.join(public_resume_dir, f))
            clean_name = f.replace(" ", "_")
            shutil.copy2(s_file, os.path.join(public_resume_dir, clean_name))

# Accomplishment & Experience -> Credentials
acc_dir = os.path.join(assests_dir, "Accomplishment")
cred_dir = os.path.join(public_assets, "credentials")
os.makedirs(cred_dir, exist_ok=True)
if os.path.exists(acc_dir):
    for f in os.listdir(acc_dir):
        s_file = os.path.join(acc_dir, f)
        if os.path.isfile(s_file):
            shutil.copy2(s_file, os.path.join(cred_dir, f))
            clean_f = f.replace(" ", "_").replace(":", "").replace("-", "_")
            shutil.copy2(s_file, os.path.join(cred_dir, clean_f))
            manifest_items.append({
                "filename": f,
                "path": f"/assets/credentials/{f}",
                "source": "Local Asset Audit",
                "project": "Credentials & Certificates",
                "usage": "Verified professional credentials and badges",
                "format": f.split(".")[-1].lower()
            })

exp_intern_dir = os.path.join(assests_dir, "Exprience", "Artificial Intelligence Intern")
if os.path.exists(exp_intern_dir):
    cert_src = os.path.join(exp_intern_dir, "Internship Certificate.pdf")
    offer_src = os.path.join(exp_intern_dir, "OFFER LETTER.pdf")
    if os.path.exists(cert_src):
        shutil.copy2(cert_src, os.path.join(cred_dir, "Codec_Technologies_AI_Intern_Certificate.pdf"))
    if os.path.exists(offer_src):
        shutil.copy2(offer_src, os.path.join(cred_dir, "Codec_Technologies_AI_Intern_Offer_Letter.pdf"))

# Write asset-manifest.json
manifest_path = os.path.join(public_assets, "asset-manifest.json")
with open(manifest_path, "w", encoding="utf-8") as f:
    json.dump({"total_assets": len(manifest_items), "assets": manifest_items}, f, indent=2)

print(f"Asset Manifest generated with {len(manifest_items)} assets.")

